db/db_impl.cc

```diff
@@ -645,6 +668,7 @@ Status DBImpl::SyncWAL() {
     need_log_dir_sync = !log_dir_synced_;
   }
 
+  TEST_SYNC_POINT("DBWALTest::SyncWALNotWaitWrite:1");
```

Comment:
@maysamyabandeh this may be old but I realized later that back() needs to be protected by the same mutex as push_back() and pop_front() too. It actually caused segfault in stress tests. I think the problem may be that deque may get copy-on-write updated, so that back() is not safely called while push_back() or pop_front() is executed even if they don't touch the element we are accessing with back(). I would continue playing safe that everything should be protected by the same mutex. In non-2pc-optimal case, it is DB mutex, and in 2pc-optimal case, it can be log_write_mutex_.